scamp/scamp-3.c

```diff
@@ -558,9 +557,17 @@ uint shm_ping(uint dest)
 uint shm_send_msg(uint dest, sdp_msg_t *msg) // Send msg AP
 {
     vcpu_t *vcpu = sv_vcpu + dest;
+    volatile uchar flag = 0;
+    event_t *e = event_new(proc_byte_set, (uint) &flag, 2);
+    if (e == NULL) {
+        sw_error(SW_OPT);
+        return RC_BUF;          // !! not the right RC
+    }
+    uint id = e->ID;
 
     sdp_msg_t *shm_msg = sark_shmsg_get();
     if (shm_msg == NULL) {
+	timer_cancel(e, id);
```

Comment:
The timer has not yet been scheduled. I think that it would be more appropriate to just free the allocated event:

event_free(e);

Reply:
Good point. I didn't spot that call as an option. (I'll need to add it to the SARK API; it's currently static in sark_event.c)